tests/Helpers/JWKFetcherTest.php

```diff
@@ -0,0 +1,195 @@
+namespace Auth0\Tests\Helpers;
+
+use \Auth0\SDK\Helpers\JWKFetcher;
+use \Auth0\Tests\TestCacheHandler;
+
+/**
+ * Class JWKFetcherTest.
+ *
+ * @package Auth0\Tests\Helpers\Cache
+ */
+class JWKFetcherTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Test that a standard JWKS path returns the first x5c.
+     *
+     * @return void
+     */
+    public function testGetJwksX5cWithoutKid()
+    {
+        $jwksFetcher = $this->getStub();
+
+        $pem = $jwksFetcher->getJwksX5c( 'https://localhost/.well-known/jwks.json' );
+
+        $this->assertNotEmpty($pem);
+
+        $pem_parts = explode( PHP_EOL, $pem );
+
+        $this->assertEquals( 4, count($pem_parts) );
+        $this->assertEquals( '-----BEGIN CERTIFICATE-----', $pem_parts[0] );
+        $this->assertEquals( '-----END CERTIFICATE-----', $pem_parts[2] );
+        $this->assertEquals( '__test_x5c_1__', $pem_parts[1] );
+    }
+
+    /**
+     * Test that a standard JWKS path returns the correct x5c when given a kid.
+     *
+     * @return void
+     */
+    public function testGetJwksX5cWithKid()
+    {
+        $jwksFetcher = $this->getStub();
+
+        $pem = $jwksFetcher->getJwksX5c( 'https://localhost/.well-known/jwks.json', '__test_kid_2__' );
+
+        $this->assertNotEmpty($pem);
+
+        $pem_parts = explode( PHP_EOL, $pem );
+
+        $this->assertEquals( 4, count($pem_parts) );
+        $this->assertEquals( '-----BEGIN CERTIFICATE-----', $pem_parts[0] );
+        $this->assertEquals( '-----END CERTIFICATE-----', $pem_parts[2] );
+        $this->assertEquals( '__test_x5c_2__', $pem_parts[1] );
+    }
+
+    /**
+     * Test that a custom JWKS path returns the correct JSON.
+     *
+     * @return void
+     */
+    public function testGetJwksX5cWithPath()
+    {
+        $jwksFetcher = $this->getStub();
+
+        $pem = $jwksFetcher->getJwksX5c( 'https://localhost/.custom/jwks.json', '__test_custom_kid__' );
+
+        $this->assertNotEmpty($pem);
+
+        $pem_parts = explode( PHP_EOL, $pem );
+
+        $this->assertEquals( 4, count( $pem_parts ) );
+        $this->assertEquals( '-----BEGIN CERTIFICATE-----', $pem_parts[0] );
+        $this->assertEquals( '-----END CERTIFICATE-----', $pem_parts[2] );
+        $this->assertEquals( '__test_custom_x5c__', $pem_parts[1] );
+    }
+
+    /**
+     * Test that a JWK with no x509 cert returns a blank array.
+     *
+     * @return void
+     */
+    public function testGetJwksX5cNoX5c()
+    {
+        $jwksFetcher = $this->getStub();
+
+        $pem = $jwksFetcher->getJwksX5c( 'https://localhost/.custom/jwks.json', '__no_x5c_test_kid_2__' );
+
+        $this->assertEmpty($pem);
+    }
+
+    /**
+     * Test that the protected getProp method returns correctly.
+     *
+     * @return void
+     */
+    public function testConvertCertToPem()
+    {
+        $class  = new \ReflectionClass(JWKFetcher::class);
+        $method = $class->getMethod('convertCertToPem');
+        $method->setAccessible(true);
```

Comment:
this looks like java reflection. I didn't know you can't test protected methods as well, normally that happens to private and static methods only. It's fine if you want to remove this as long as it's being black-box tested somewhere else.

Reply:
I wanted to be safe here and make sure it's creating the correct chunks, etc.